test/test_utils.py

```diff
@@ -184,5 +187,124 @@ def test_delete_none(self):
         self.assertEqual(delete_none(input), truth)
 
 
+@pytest_asyncio.fixture(scope="function", autouse=True)
+async def client():
+    """
+    A fixture to mock the async client object.
+    """
+    async with AsyncClient() as mock_client:
+        with mock.patch("gradio.utils.client", mock_client):
+            yield
+
+
+class TestRequest:
+    @pytest.mark.asyncio
+    async def test_get(self):
+        client_response: Request = await Request(
+            method=Request.Method.GET,
+            url="http://headers.jsontest.com/",
+        )
+        validated_data = client_response.get_validated_data()
+        assert client_response.is_valid() is True
+        assert validated_data["Host"] == "headers.jsontest.com"
+
+    @pytest.mark.asyncio
+    async def test_post(self):
+        client_response: Request = await Request(
+            method=Request.Method.POST,
+            url="https://reqres.in/api/users",
+            json={"name": "morpheus", "job": "leader"},
+        )
+        validated_data = client_response.get_validated_data()
+        assert client_response.status == 201
+        assert validated_data["job"] == "leader"
+        assert validated_data["name"] == "morpheus"
+
+    @pytest.mark.asyncio
+    async def test_validate_with_model(self):
+        class TestModel(BaseModel):
+            name: str
+            job: str
+            id: str
+            createdAt: str
+
+        client_response: Request = await Request(
+            method=Request.Method.POST,
+            url="https://reqres.in/api/users",
+            json={"name": "morpheus", "job": "leader"},
+            validation_model=TestModel,
+        )
+        assert isinstance(client_response.get_validated_data(), TestModel)
+
+    @pytest.mark.asyncio
+    async def test_validate_and_fail_with_model(self):
+        class TestModel(BaseModel):
+            name: Literal[str] = "John"
```

Comment:
Type should be "str" instead of "Literal[str]"

Reply:
Thx for catching, I've missed that, I shall update in my current PR